### ISSUE TITLE
Windows support for getting the current brightness level

### ISSUE THREAD

**Issue by sindresorhus** [kevva/brightness#8]
Currently only setting the brightness is supported.

**Comment by sondreb** [comment]
Did some research to find a solution:

It can be retrieved using WMI (Windows Management Instrumentation) with this operation`from Windows PowerShell ([documentation](https://msdn.microsoft.com/en-us/library/aa394536(v=vs.85).aspx)):

```
Get-Ciminstance -Namespace root/WMI -ClassName WmiMonitorBrightness
```
Returns a structure with a property named "CurrentBrightness" that goes from 0 to 100 (uint8).

Only runs on Vista or higher.

Setting the brightness can be done with ([documentation](https://msdn.microsoft.com/en-us/library/aa394549(v=vs.85).aspx)):

```
$monitor = Get-WmiObject -Namespace root\wmi -Class WmiMonitorBrightnessMethods
$monitor.wmisetbrightness(0, 25)
```

There is additionally a Windows API (GetMonitorBrightness) that gives the same, also only runs on Vista or higher:

https://msdn.microsoft.com/en-us/library/ms775205.aspx

I presume this is the way NirCmd (used by Brightness) does it's set operation:

http://www.nirsoft.net/vc/change_screen_brightness.html

**Comment by sindresorhus** [comment]
> Get-Ciminstance -Namespace root/WMI -ClassName WmiMonitorBrightness

Is there a way to do this from cmd.exe? I know you can use [WMI like a normal command](https://github.com/sindresorhus/kill-tabs/blob/9ae45fe7e06c6758b85b856ba9436db8ae372b2e/win.js#L6).

**Comment by sondreb** [comment]
Here is code to make it work. This WMI client supports Linux as well, so it might be an option to use WMI for both Windows and Linux:

```javascript
var WmiClient = require('wmi-client');

var wmi = new WmiClient({
    host: 'localhost',
    namespace: '\\\\root\\WMI' // Important, defaults to '\\root\cimv2'
});

// Doing select ONLY on CurrentBrightness as oppose to * or multiple columns, returns empty array!?
wmi.query('SELECT * FROM WmiMonitorBrightness', function (err, result) {
    // There might be multiple monitors... I presume that's why it returns an array?
    var currentBrightness = result[0].CurrentBrightness;
    console.log('CurrentBrightness: ' + currentBrightness + "%.");
});

```

**Comment by sindresorhus** [comment]
@sondreb Awesome! Interested in doing a pull request?

**Comment by kevva** [comment]
This is great @sondreb, PR is super welcome!

**Comment by sondreb** [comment]
@sindresorhus / @kevva I will do a PR, just wanted to mention it here so nobody else does it and duplicates efforts. Will try to get it done tonight or tomorrow. I might do it in two phases, first is adding read support (for Windows) and then perhaps consider using WMI to set brightness as well.

**Comment by sondreb** [comment]
There appears to be a considerable delay when executing powershell.exe from node.js, the same is true from a regular cmd.exe in Windows. I had a working solution using PowerShell, but I changed to wmic which is a command tool for WMI operations on Windows (used by wmi-client package as well).

wmic solution is not working, as I'm unable to find the proper syntax for parameters. I got it almost complete, but until then, nircmd can be used for setting. Should be updated as soon as I find a solution for wmic, as that should probably be better and reduce dependencies a little bit. Already spent the whole day investigating this thing.

Made a [PR](https://github.com/kevva/brightness/pull/10) and additionally I made a [win-brightness](https://www.npmjs.com/package/win-brightness) package. I hope it was OK I took some code from you @kevva, added some links in the readme on the repo.

At least I learnt a lot more about WMI, wmic and PowerShell...
